### PR TITLE
feat(gcf-utils)!: Use Secret Manager as opposed to KMS + GCS

### PR DESCRIPTION
**Pull request by orthros**
Now uses SecretManager to store Probot configs. This obviates the DIY
solution we had for storing these secrets.

### CODE REVIEW

**Comment by orthros** [comment]
Overall, this is functional, but needs to have the `bin` script updated to be simpler (and not use KMS and GCS). Will be looking at that next

File: packages/gcf-utils/src/bin/genkey-util.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  webhookSecret: string
+): Promise<Options> {
+  let keyContent = '';
+  try {
+    keyContent = fs.readFileSync(keyfile, 'utf8');
+  } catch (e) {
+    throw Error(`Error reading file: ${keyfile}`);
+  }
+
+  const blob: Options = {
```
Comment (bcoe) [inline packages/gcf-utils/src/bin/genkey-util.ts:31]: I think you could make this a one liner with:

```js
return {
    cert: keyContent,
    id,
    secret: webhookSecret,
}  as Options
```

File: packages/gcf-utils/test/gcf-utils.ts [hunk 2]
```diff
@@ -212,4 +214,98 @@ describe('GCFBootstrapper', () => {
...
   });
+
+  describe('getProbotConfig', () => {
+    let bootstrapper: GCFBootstrapper;
+    let secretClientStub: v1.SecretManagerServiceClient;
+    let secretsStub: sinon.SinonStub;
+    let secretVersionNameStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      secretClientStub = new v1.SecretManagerServiceClient();
```
Comment (bcoe) [inline packages/gcf-utils/test/gcf-utils.ts:225]: I just realized that @alexander-fenster's refactor to make gRPC delay initialization made stubbing _way_ easier, this is great.

Comment (orthros) [reply packages/gcf-utils/test/gcf-utils.ts:225]: Yep! It is so so so nice to use

File: packages/gcf-utils/src/gcf-utils.ts [hunk 3]
```diff
@@ -57,36 +61,25 @@ export class GCFBootstrapper {
...
-      ciphertext: contentsBuffer,
+  async getProbotConfig(): Promise<Options> {
+    const name = this.getLatestSecretVersionName();
+    const [version] = await this.secretsClient.accessSecretVersion({
+      name: name,
     });
-
-    const config = JSON.parse(result.plaintext!.toString());
+    // Extract the payload as a string.
+    const payload = version?.payload?.data?.toString() || '';
```
[the remark this answers is not in the document]
Comment (orthros) [reply packages/gcf-utils/src/gcf-utils.ts:81]: Done; also added some tests for it

File: packages/gcf-utils/src/bin/genkey-util.ts (hunk 1 above)
Comment (orthros) [reply packages/gcf-utils/src/bin/genkey-util.ts]: Done

**Comment by orthros** [comment]
Latest commits add a new script to migrate from GCS to SecretManager. This script will not work right now, however as `publish` bot has a secret already named "publish" in our project. 

Once that secret has been migrated, we can run the script and merge this PR.

**Pull request merged**